fvm/evm/handler/handler.go

```diff
@@ -134,45 +134,75 @@ func (h *ContractHandler) LastExecutedBlock() *types.Block {
 }
 
 // RunOrPanic runs an rlp-encoded evm transaction and
-// collects the gas fees and pay it to the coinbase address provided.
-func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, coinbase types.Address) {
+func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, gasFeeCollector types.Address) {
 	// capture open tracing span
 	defer h.backend.StartChildSpan(trace.FVMEVMRun).End()
 
-	res, err := h.run(rlpEncodedTx, coinbase)
+	// capture coinbase init balance
```

Comment:
why do we do this in the contract handler? it feels a bit awkward place for it? why not in the emulator?

Reply:
It was harder to do it on the Emulator side because of edge cases we could have without panics on status of execution

Reply:
Another reason was anyway the handler deals with a lot of high-level tasks like emitting events, etc. This seems like a high-level task to refund gas.